### PR TITLE
CU-86b1g1mye_Generar-un-enum-de-FieldNames-para-validar-que-el-Field-…

### PR DESCRIPTION
**Pull request by ptorres-prowide**
…exista

## Summary by CodeRabbit

- **New Features**
	- Introduced `SRU2023FieldEnum`, enhancing field management by providing a structured enumeration of field names for improved data handling and validation.
	- Introduced `FieldEnum`, offering a comprehensive list of field constants to streamline reference and enhance code readability.

- **Documentation**
	- Updated `CHANGELOG.md` to reflect the changes and enhancements in version `9.4.18`.

- **Tests**
	- Added a suite of unit tests for the `FieldEnum` class to ensure functionality and handle edge cases effectively.

### CODE REVIEW

File: src/generated/java/com/prowidesoftware/swift/model/field/FieldEnum.java [hunk 1]
```diff
@@ -0,0 +1,738 @@
...
+    F434,
+    F443,
+    F451,
+    F461,
+    F503,
+    F533,
+    F619,
+    F999;
+
+    public String getFieldName() {
```
Comment (zubri) [inline src/generated/java/com/prowidesoftware/swift/model/field/FieldEnum.java:726]: for convention when method is not an accessor we do not add the "get" prefix. This should be fieldName()

File: src/generated/java/com/prowidesoftware/swift/model/field/FieldEnum.java [hunk 2]
```diff
@@ -0,0 +1,738 @@
...
+        return name().substring(1); // Skips the first character 'F' and returns the rest of the name
+    }
+
+    public static FieldEnum fromCode(String code) {
```
Comment (zubri) [inline src/generated/java/com/prowidesoftware/swift/model/field/FieldEnum.java:730]: This method is not needed, as it is the same as the valueOf already provided by the Java enum

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,8 @@
 # Prowide Core - CHANGELOG
 
+#### 9.4.18 - SNAPSHOT
+  * Added new SRU2023FieldEnum with all the available field names
```
Comment (zubri) [inline CHANGELOG.md:4]: class was renamed

**Review by zubri — CHANGES_REQUESTED** [review]

File: src/generated/java/com/prowidesoftware/swift/model/field/FieldEnum.java (hunk 1 above)
Comment (ptorres-prowide) [reply src/generated/java/com/prowidesoftware/swift/model/field/FieldEnum.java:726]: Ok

File: src/generated/java/com/prowidesoftware/swift/model/field/FieldEnum.java (hunk 2 above)
Comment (ptorres-prowide) [reply src/generated/java/com/prowidesoftware/swift/model/field/FieldEnum.java:730]: Ok

File: CHANGELOG.md (hunk 3 above)
Comment (ptorres-prowide) [reply CHANGELOG.md:4]: Ok

File: src/generated/java/com/prowidesoftware/swift/model/field/FieldEnum.java (hunk 2 above)
Comment (zubri) [reply src/generated/java/com/prowidesoftware/swift/model/field/FieldEnum.java:730]: + javadoc
fromFieldName(String fieldName)

**Review by zubri — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,8 @@
 # Prowide Core - CHANGELOG
 
+#### 9.4.18 - SNAPSHOT
```
[the remark this answers is not in the document]
Comment (zubri) [reply CHANGELOG.md:3]: no needed
